src/engine/BMInterfaceTournament.php

```diff
@@ -0,0 +1,1101 @@
+                TRUE,
+                array(),
+                $tournament->tournamentId,
+                $gameData['roundNumber']
+            );
+
+            // add game number to $this->gameIdArrayArray
+            if (is_array($interfaceResponse) && array_key_exists('gameId', $interfaceResponse)) {
+                $gameIdArrayArray = $tournament->gameIdArrayArray;
+                $gameIdArrayArray[$gameData['roundNumber'] - 1][] = $interfaceResponse['gameId'];
+                $tournament->gameIdArrayArray = $gameIdArrayArray;
+            }
+        }
+    }
+
+    /**
+     * Most of the tournament saving logic
+     *
+     * @param BMTournament $tournament
+     */
+    protected function save_basic_tournament_parameters(BMTournament $tournament) {
+        $query = 'UPDATE tournament '.
+                 'SET status_id = '.
+                 '        (SELECT id FROM tournament_status WHERE name = :status),'.
+                 '    tournament_state = :tournament_state,'.
+                 '    round_number = :round_number '.
+                 'WHERE id = :tournament_id;';
+        $parameters = array(':status' => $this->get_tournament_status($tournament),
+                            ':tournament_state' => $tournament->tournamentState,
+                            ':round_number' => $tournament->roundNumber,
+                            ':tournament_id' => $tournament->tournamentId);
+        self::$db->update($query, $parameters);
+    }
+
+    /**
+     * Logic to save player parameters
+     *
+     * @param BMTournament $tournament
+     */
+    protected function save_player_parameters(BMTournament $tournament) {
+        // james: In adding the ability to save button selection, I've taken the shortcut
+        //        of assuming that only one button is being selected at the moment
+
+        foreach ($tournament->playerIdArray as $playerIdx => $playerId) {
+            $query = 'UPDATE tournament_player_map '.
+                     'SET position = :position,'.
+                     '    remain_count = :remain_count,'.
+                     '    button_id = :button_id '.
+                     'WHERE tournament_id = :tournament_id '.
+                     'AND player_id = :player_id;';
+            $parameters = array(':position' => $playerIdx + 1,
+                                ':remain_count' => $tournament->remainCountArray[$playerIdx],
+                                ':button_id' => $tournament->buttonIdArrayArray[$playerId][0],
+                                ':tournament_id' => $tournament-> tournamentId,
+                                ':player_id' => $playerId);
+            self::$db->update($query, $parameters);
+        }
+    }
+
+    /**
+     * Force a tournament to update
+     *
+     * @param int $tournamentId
+     */
+    protected function update_tournament($tournamentId) {
+        if ($this->isTest) {
+            return;
+        }
```

Comment:
This block breaks responder testing of tournaments, because you need the combination of load + save to progress a tournament through states, e.g. from JOIN to SHUFFLE and onward.

I suspect you can simply remove it --- there aren't any `isTest` checks in `BMInterfaceGame.php`, so i'm not sure why you'd need them here.

Reply:
My suspicion is that I added this to avoid doing responder testing when we had no responder testing for tournaments. I've removed it now.